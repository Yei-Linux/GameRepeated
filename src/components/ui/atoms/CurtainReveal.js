import CurtainRevealBox from 'react-curtain-reveal'

const CurtainReveal = ({
  isRevealed = false,
  revealCheck = () => {},
  onReveal = () => {},
  onHide = () => {},
  children,
}) => {
  return (
    <CurtainRevealBox
      revealCheck={revealCheck}
      isRevealed={isRevealed}
      onReveal={onReveal}
      onHide={onHide}
      styleConfig={{
        curtain: {
          borderRadius: 0,
        },
        leftPanel: {
          background:
            'linear-gradient(90deg, rgba(34,193,195,1) 0%, rgba(253,187,45,1) 100%)',
        },
        rightPanel: {
          background:
            'linear-gradient(270deg, rgba(34,193,195,1) 0%, rgba(253,187,45,1) 100%)',
        },
        content: {
          backgroundColor: 'white',
        },
      }}
      animationConfig={{
        variant: 'linear',
        speed: 5,
      }}
    >
      {children}
    </CurtainRevealBox>
  )
}

export default CurtainReveal
