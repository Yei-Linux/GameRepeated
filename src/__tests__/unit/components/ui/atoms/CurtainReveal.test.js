import { render } from '@testing-library/react'
import CurtainReveal from '../../../../../components/ui/atoms/CurtainReveal'

describe('The CurtainReveal component', () => {
  const defaultProps = {
    isRevealed: true,
    revealCheck: () => false,
    onReveal: jest.fn(),
    onHide: jest.fn(),
    children: <p>TextUI</p>,
  }

  const component = (props = defaultProps) => <CurtainReveal {...props} />

  const setupComponent = (props = defaultProps) => render(component(props))

  it('Should validate isRevealed prop', async () => {
    const { container, rerender } = setupComponent()

    const curtainReveal = container.querySelector(
      '.makeStyles-curtain_wrapper-2'
    )

    expect(curtainReveal).toBeInTheDocument()

    rerender(component({ ...defaultProps, isRevealed: true }))

    const curtainRerender = container.querySelector(
      '.makeStyles-curtain_wrapper-2'
    )

    expect(curtainRerender).not.toBeInTheDocument()
  })

  it('Should render CurtainReveal component', () => {
    const { asFragment } = setupComponent()

    expect(asFragment()).toMatchSnapshot()
  })
})
