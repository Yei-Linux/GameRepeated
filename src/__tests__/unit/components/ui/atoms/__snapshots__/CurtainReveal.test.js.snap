// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CurtainReveal component Should render CurtainReveal component 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-curtain-23 "
  >
    <div
      class="makeStyles-curtain_wrapper-24"
    >
      <button
        class="makeStyles-curtain_trigger_box-29 makeStyles-hide-32"
      />
      <div
        class="makeStyles-curtain_panel-25 makeStyles-curtain_panel_left-26 makeStyles-reveal_left-30"
      />
      <div
        class="makeStyles-curtain_content-28"
      >
        <p>
          TextUI
        </p>
      </div>
      <div
        class="makeStyles-curtain_panel-25 makeStyles-curtain_panel_right-27 makeStyles-reveal_right-31"
      />
    </div>
  </div>
</DocumentFragment>
`;
